File: app/src/main/java/com/luoyang/myandroidxstudy/api/MyAndroidxStudyServer.java
package com.luoyang.myandroidxstudy.api;

import com.google.gson.JsonObject;
import com.luoyang.myandroidxstudy.bean.Find;
import com.luoyang.myandroidxstudy.bean.User;
import com.luoyang.myandroidxstudy.net.NetConstant;
import retrofit2.Call;
import retrofit2.http.GET;

import java.util.List;

/**
 * 网络请求服务
 *
 * @author LuoYang
 */
public interface MyAndroidxStudyServer {
    @GET(NetConstant.FIND)
    Call<ResponseData<List<Find>>> find();

    @GET(NetConstant.BOUGHT)
    Call<ResponseData<JsonObject>> bought();

    @GET(NetConstant.ME)
    Call<ResponseData<User>> getMe();
}

File: app/src/main/java/com/luoyang/myandroidxstudy/api/MyAndroidxStudyServer.kt
package com.luoyang.myandroidxstudy.api

import com.google.gson.JsonObject
import com.luoyang.myandroidxstudy.bean.Find
import com.luoyang.myandroidxstudy.bean.User
import com.luoyang.myandroidxstudy.net.NetConstant
import retrofit2.Call
import retrofit2.http.GET

/**
 * 网络请求服务
 */
interface MyAndroidxStudyServer {
    @GET(NetConstant.FIND)
    fun find(): Call<ResponseData<List<Find>>>

    @GET(NetConstant.BOUGHT)
    fun bought(): Call<ResponseData<JsonObject>>

    @GET(NetConstant.ME)
    fun getMe(): Call<ResponseData<User>>
}